Mathlib/Geometry/Manifold/IntegralCurve/UniformTime.lean

```diff
@@ -0,0 +1,209 @@
+/-
+Copyright (c) 2023 Winston Yin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Winston Yin
+-/
+import Mathlib.Geometry.Manifold.IntegralCurve.ExistUnique
+
+/-!
+# Uniform time lemma for the global existence integral curves
+
+## Main results
+
+* `exists_isIntegralCurve_of_isIntegralCurveOn`: If there exists `ε > 0` such that the local
+integral curve at each point `x : M` is defined at least on an open interval `Ioo (-ε) ε`, then
+every point on `M` has a global integral curve passing through it.
+
+## Reference
+
+* [Lee, J. M. (2012). _Introduction to Smooth Manifolds_. Springer New York.][lee2012]
+
+## Tags
+
+integral curve, vector field, global existence
+-/
+
+open scoped Topology
+
+open Function Set
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+  {H : Type*} [TopologicalSpace H] {I : ModelWithCorners ℝ E H}
+  {M : Type*} [TopologicalSpace M] [ChartedSpace H M] [SmoothManifoldWithCorners I M] [T2Space M]
+  {γ γ' : ℝ → M} {v : (x : M) → TangentSpace I x} {s s' : Set ℝ} {t₀ : ℝ}
+
+/-- This is the uniqueness theorem of integral curves applied to a real-indexed family of integral
+  curves with the same starting point. -/
+lemma eqOn_of_isIntegralCurveOn_Ioo [BoundarylessManifold I M]
+    (hv : ContMDiff I I.tangent 1 (fun x ↦ (⟨x, v x⟩ : TangentBundle I M))) {x : M}
+    (γ : ℝ → ℝ → M) (hγx : ∀ a, γ a 0 = x) (hγ : ∀ a, IsIntegralCurveOn (γ a) v (Ioo (-a) a))
+    {a a' : ℝ} (hpos : 0 < a') (hle : a' ≤ a) :
+    EqOn (γ a') (γ a) (Ioo (-a') a') := by
```

Comment:
I don't know the convention very well either, but I don't think `epsilon` makes sense here: `a` is any real number, not a "small" one. One could use `t` instead of `a` (no strong opinion).

Right now, this lemma is just used once (in one proof, twice). I'm not sure if there will be further uses. Perhaps, when speaking about the existence of geodesics... (which are a bit away, as mathlib is missing Riemannian metrics, hence the exponential map). What do you think @winstonyin?

Reply:
But at the place where it is used, the same assumption is in place, so the weakening could propagate.

Reply:
:hammer_and_pick:

Reply:
I just pushed a commit doing this. Overall, the code gets a bit longer. What do you think?